tests/publisher_test.py

```diff
@@ -404,3 +404,90 @@ def testUsesUnionOfMetaKeys(self):
+                        'official=1.0', "metadata={'info': '1', \
+                        'bar': 'foo', 'more_info': '2'}", 'value=non',
+                        'sample_uri=33'])
+
+    def testConstructSample(self):
+      no_meta = self.mock_db._ConstructSample(self.no_meta)
+      empty_meta = self.mock_db._ConstructSample(self.empty_meta)
+      with_meta = self.mock_db._ConstructSample(self.with_meta)
+
+      self.assertEqual(no_meta,
+                       'perfkitbenchmarker,metric=3,official=47.0,\
+                       owner=Rackspace,run_uri=5rtw,test=testa,\
+                       unit=us,sample_uri=5r value=non 123000000000')
+      self.assertEqual(empty_meta,
+                       'perfkitbenchmarker,metric=2,official=14.0,\
+                       owner=Rackspace,run_uri=bba3,test=testb,unit=MB,\
+                       sample_uri=bb value=non 55000000000')
+      self.assertEqual(with_meta,
+                       'perfkitbenchmarker,metric=1,official=1.0,\
+                       owner=Rackspace,run_uri=323,test=testc,unit=MB,\
+                       sample_uri=33,info=1,bar=foo,\
+                       more_info=2 value=non 123000000000')
+
+    def testAggregationOfSamples(self):
+      sample_data = [{'test': 'testc', 'metric': '1', 'official': 1.0,
+                     'value': 'non', 'unit': 'MB', 'owner': 'Rackspace',
+                      'run_uri': '323', 'sample_uri': '33', 'timestamp': 123,
+                      'metadata':
+                      {'info': '1', 'more_info': '2', 'bar': 'foo'}},
+                     {'test': 'testb', 'metric': '2', 'official': 14.0,
+                      'value': 'non', 'unit': 'MB', 'owner': 'Rackspace',
+                      'run_uri': 'bba3', 'sample_uri': 'bb', 'timestamp': 55,
+                      'metadata': {}},
+                     {'test': 'testa', 'metric': '3', 'official': 47.0,
+                     'value': 'non', 'unit': 'us', 'owner': 'Rackspace',
+                      'run_uri': '5rtw', 'sample_uri': '5r', 'timestamp': 123}]
+      formated_samples = []
+      expected_output = ('perfkitbenchmarker,metric=1,official=1.0,'
+                         'owner=Rackspace,run_uri=323,test=testc,unit=MB,'
+                         'sample_uri=33,info=1,bar=foo,more_info=2 value=non '
+                         '123000000000\nperfkitbenchmarker,metric=2,'
+                         'official=14.0,owner=Rackspace,run_uri=bba3,'
+                         'test=testb,unit=MB,sample_uri=bb value=non '
+                         '55000000000\nperfkitbenchmarker,metric=3,'
+                         'official=47.0,owner=Rackspace,run_uri=5rtw,'
+                         'test=testa,unit=us,sample_uri=5r '
+                         'value=non 123000000000')
+      for data in sample_data:
+        formated_samples.append(self.mock_db._ConstructSample(sample))
+        self.assertEqual(formated_samples, expected_output)
```

Comment:
I'm not sure I understand how this test works.

Wouldn't this be calling an assertEqual between the `formatted_samples` list and the string `expected_output`? Also, the list `formatted_samples` changes in every iteration as the samples are added but the string `expected output` remains constant. This test would fail every time.